### PR TITLE
make getResponseTime deterministic

### LINKED ISSUE #3430: Calling Reply.getResponseTime() multiple times results in different times
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

3.22.0

### Plugin version

_No response_

### Node.js version

14.18.1

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

11.6.1

### Description

Calling Reply.getResponseTime() multiple times results in different values. Optimally the response time should be set at some point and the function simply return that value.

### Steps to Reproduce

The function uses Date.now() to calculate the value every time it is called.

### Expected Behavior

The response time should be the same for one requests independent of when you are calling the getResponseTime function.

### PR DESCRIPTION
**Pull request by thomaschaaf**
fixes #3430

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by zekth — APPROVED** [review]
Good catch!

**Review by delvedor — APPROVED** [review]
LGTM

**Review by climba03003 — APPROVED** [review]
LGTM.

**Review by mcollina — CHANGES_REQUESTED** [review]
Good work! This is also a breaking change.

The previous behavior had `getResponseTime()` return the time it took _so far_ while the request was in flight. Could you restore this? It's probably an else-clause.

File: lib/reply.js [hunk 1]
```diff
@@ -612,6 +613,7 @@ function setupResponseListeners (reply) {
   reply[kReplyStartTime] = now()
 
   const onResFinished = err => {
+    reply[kReplyEndTime] = now()
```
Comment (mcollina) [inline lib/reply.js:616]: set this only if `this[kReplyStartTime]` is set. Getting the time is relatively expensive.

Comment (thomaschaaf) [reply lib/reply.js:616]: I don't understand how it could not be set. As it is set in the parent function.

Comment (mcollina) [reply lib/reply.js:616]: Ah sorry, I didn't see it!

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/reply.js [hunk 2]
```diff
@@ -311,8 +312,8 @@ Reply.prototype.callNotFound = function () {
 Reply.prototype.getResponseTime = function () {
   let responseTime = 0
 
-  if (this[kReplyStartTime] !== undefined) {
-    responseTime = now() - this[kReplyStartTime]
+  if (this[kReplyStartTime]) {
```
Comment (jsumners) [inline lib/reply.js:315]: Why the coercion?

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Pull request merged**